packages/actor-init-sparql/lib/ActorInitSparql-browser.ts

```diff
@@ -31,6 +36,49 @@ export class ActorInitSparql extends ActorInit implements IActorInitSparqlArgs {
     super(args);
   }
 
+  /**
+   * Create a copy of the given operation in which all given bindings are applied.
+   * The bindings are applied to all quad patterns and path expressions.
+   *
+   * @param {Operation} operation An operation.
+   * @param {Bindings} initialBindings Bindings to apply.
+   * @return {Operation} A copy of the given operation where all given bindings are applied.
+   */
+  public static applyInitialBindings(operation: Algebra.Operation, initialBindings: Bindings): Algebra.Operation {
+    const copiedOperation: Algebra.Operation = <any> {};
+    for (const key of Object.keys(operation)) {
+      if (Array.isArray(operation[key])) {
+        if (key === 'variables') {
+          copiedOperation[key] = operation[key].filter(
+            (variable: RDF.Variable) => initialBindings.keySeq().indexOf(termToString(variable)) < 0);
```

Comment:
This can change the meaning of a query in case of Group By and Values queries though.

Reply:
Indeed, that was intentional.
But I'm open to other suggestions.
Do you think it would make sense to either not remove the variables in those cases, or throw an error?